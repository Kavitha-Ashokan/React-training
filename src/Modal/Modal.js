import React from "react";
import Modal from "react-modal";


function Example({ show, onHide, values}) {
    console.log(values);
  return (
    <Modal className="modal-box" isOpen={show} onRequestClose={onHide}>
      <h2>Shipping</h2>
      <h4>{JSON.stringify(values)}</h4>
      <button onClick={onHide}>close</button>
    </Modal>
  );
}

export default Example;