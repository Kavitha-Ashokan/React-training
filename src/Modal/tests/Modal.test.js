import React from 'react';
import renderer from "react-test-renderer";
import Modal from '../Modal';

const mockDetails={
modalDetails:{
    show:true,
    onHide:jest.fn(),
    values:{}
}
}

it("should render and match the snapshot",()=>{
    const tree = renderer.create( <Modal/> ).toJSON();
    expect(tree).toMatchSnapshot();
})
it("check props correctly rendered",()=>{
    const tree = renderer.create( <Modal modal={mockDetails}/> ).toJSON();
    expect(tree).toMatchSnapshot();
})
