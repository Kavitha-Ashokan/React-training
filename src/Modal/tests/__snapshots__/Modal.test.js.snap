// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check props correctly rendered 1`] = `null`;

exports[`should render and match the snapshot 1`] = `null`;
